lib/core/storages/memory.ts

```diff
@@ -15,6 +16,10 @@ export class MemoryStorage<T> implements IStorage<T> {
   async rmd(id: string, func: (item: T) => boolean): Promise<void> {
     const item = this.items[id];
 
+    if (!item) {
+      return Promise.reject(new ResonateError(ErrorCodes.NOT_FOUND, `Not found`));
```

Comment:
Can we move this to the store instead of the storage? That way it will apply to all underlying storages